deegree-core/deegree-core-featureinfo/src/main/java/org/deegree/featureinfo/parsing/DefaultFeatureInfoParser.java

```diff
@@ -0,0 +1,239 @@
+ You should have received a copy of the GNU Lesser General Public License
+ along with this library; if not, write to the Free Software Foundation, Inc.,
+ 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+
+ Contact information:
+
+ lat/lon GmbH
+ Aennchenstr. 19, 53177 Bonn
+ Germany
+ http://lat-lon.de/
+
+ Department of Geography, University of Bonn
+ Prof. Dr. Klaus Greve
+ Postfach 1147, 53001 Bonn
+ Germany
+ http://www.geographie.uni-bonn.de/deegree/
+
+ Occam Labs UG (haftungsbeschränkt)
+ Godesberger Allee 139, 53175 Bonn
+ Germany
+
+ e-mail: info@deegree.org
+ ----------------------------------------------------------------------------*/
+package org.deegree.featureinfo.parsing;
+
+import static org.deegree.commons.tom.primitive.BaseType.STRING;
+import static org.deegree.commons.xml.stax.XMLStreamUtils.nextElement;
+import static org.deegree.commons.xml.stax.XMLStreamUtils.skipElement;
+import static org.deegree.gml.GMLInputFactory.createGMLStreamReader;
+import static org.deegree.gml.GMLVersion.GML_2;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import org.deegree.commons.tom.gml.property.Property;
+import org.deegree.commons.tom.gml.property.PropertyType;
+import org.deegree.commons.xml.XMLParsingException;
+import org.deegree.commons.xml.stax.XMLStreamUtils;
+import org.deegree.cs.exceptions.UnknownCRSException;
+import org.deegree.feature.FeatureCollection;
+import org.deegree.feature.GenericFeature;
+import org.deegree.feature.GenericFeatureCollection;
+import org.deegree.feature.property.SimpleProperty;
+import org.deegree.feature.types.DynamicAppSchema;
+import org.deegree.feature.types.GenericFeatureType;
+import org.deegree.feature.types.property.SimplePropertyType;
+import org.deegree.gml.GMLStreamReader;
+import org.slf4j.Logger;
+
+/**
+ * Responsible for parsing 'feature collections', even if they are broken (eg. ESRI or UMN mapserver feature info
+ * responses). Used by the WMS and WMTS clients. Currently, ESRI, UMN mapserver, mywms and normal GML2 feature
+ * collections are supported.
+ * 
+ * @author <a href="mailto:schmitz@occamlabs.de">Andreas Schmitz</a>
+ * @author last edited by: $Author: stranger $
+ * 
+ * @version $Revision: $, $Date: $
+ */
+public class DefaultFeatureInfoParser implements FeatureInfoParser {
+
+    private static final Logger LOG = getLogger( DefaultFeatureInfoParser.class );
+
+    private static final XMLInputFactory XML_FACTORY = XMLInputFactory.newInstance();
+    
+    @Override
+    public FeatureCollection parseAsFeatureCollection( InputStream inputStream, String csvLayerNames )
+                            throws XMLStreamException {
+        XMLStreamReader xmlReader = XML_FACTORY.createXMLStreamReader( inputStream );
```

Comment:
As a remark, we have a security optimized way of getting a safe at https://github.com/deegree/deegree3/blob/master/deegree-core/deegree-core-commons/src/main/java/org/deegree/commons/xml/stax/XMLInputFactoryUtils.java#L60